### DISCUSSION TITLE
Ignore non-existing datasource

### DISCUSSION THREAD

**Discussion by ghost** [hairyhenderson/gomplate#1537] [Q&A] [answered]
Hello,

We are trying to generate providers and resources list for Terraform based on two sources [concatenation](1168)
1. CI defined variable
2. AWS Secrets Manager secret

AWS SM secret is created by this Terraform code and it means that at the initial run secret doesn't exist yet and execution fails:
```bash
gomplate -f main.tf -o - -d secrets_regions=aws+sm:secrets_regions
```
> 20:36:18 ERR  error="failed to render template main.tf: template: main.tf:13:53: executing \"main.tf\" at <ds \"secrets_regions\">: error calling ds: Couldn't read datasource 'secrets_regions': reading aws+sm source \"secrets_regions\": ResourceNotFoundException: Secrets Manager can't find the specified secret."

Is there a way to ignore the error of non-existing datasource?

**Comment by hairyhenderson** [comment, marked as answer]
Hi @air3ijai, thanks for taking the time to post this.

_(FYI I edited your post to fix a broken link to #1168)_

I think what you're looking for is the [`datasourceReachable`](https://docs.gomplate.ca/functions/data/#datasourcereachable) function, which is designed for this type of situation.

Another option is to use [`datasourceExists`](https://docs.gomplate.ca/functions/data/#datasourceexists), and simply not specify the `secrets_regions` datasource on the first run (if you know the value won't exist).

See the docs for examples on how to use these in an `if`/`else` block.
